### PR TITLE
Improve RuboCop rules (compatibility to Code Climate)

### PR DESCRIPTION
**Pull request by ykzts**
Code Climate is using [RuboCop v0.46.0](https://github.com/codeclimate/codeclimate-rubocop/blob/08f8de84ebfb39caa96391e23816877278f6441c/Gemfile.lock#L38).

Change several rules to maintain compatibility.

### before

```console
$ bundle exec rubocop
Inspecting 251 files
CC.C......C..C.C.......C..........C.........C..C........C.CC.....C..............CC...CC....CC......C.C..C.....CC........C.CC..C......CC..CCCCC.CC.C.CCCCC.C.CCC.CCC........C.....CC...C.....C.C.C.......CC.....C....C....C.........C.......CC..........C.CC

Offenses:

# ...

251 files inspected, 131 offenses detected
```

### after

```console
$ bundle exec rubocop
Inspecting 251 files
.....................................................................................................C..C.....C...................................................................C...C.....C.C................C...........................................

Offenses:

app/lib/atom_serializer.rb:3:1: C: Class has too many lines. [257/200]
class AtomSerializer ...
^^^^^^^^^^^^^^^^^^^^
app/lib/feed_manager.rb:97:3: C: Perceived complexity for filter_from_home? is too high. [11/10]
  def filter_from_home?(status, receiver_id)
  ^^^
app/lib/provider_discovery.rb:7:5: C: Perceived complexity for discover_provider is too high. [11/10]
    def discover_provider(url, options = {})
    ^^^
app/services/fetch_link_card_service.rb:47:3: C: Cyclomatic complexity for attempt_oembed is too high. [16/15]
  def attempt_oembed(card, url)
  ^^^
app/services/fetch_link_card_service.rb:47:3: C: Perceived complexity for attempt_oembed is too high. [14/10]
  def attempt_oembed(card, url)
  ^^^
app/services/follow_remote_account_service.rb:14:3: C: Cyclomatic complexity for call is too high. [16/15]
  def call(uri, redirected = nil)
  ^^^
app/services/follow_remote_account_service.rb:14:3: C: Perceived complexity for call is too high. [17/10]
  def call(uri, redirected = nil)
  ^^^
app/services/process_feed_service.rb:22:3: C: Class has too many lines. [209/200]
  class ProcessEntry ...
  ^^^^^^^^^^^^^^^^^^
app/services/process_feed_service.rb:44:5: C: Perceived complexity for create_status is too high. [15/10]
    def create_status
    ^^^
app/services/process_interaction_service.rb:10:3: C: Cyclomatic complexity for call is too high. [25/15]
  def call(envelope, target_account)
  ^^^
app/services/process_interaction_service.rb:10:3: C: Perceived complexity for call is too high. [16/10]
  def call(envelope, target_account)
  ^^^
app/services/update_remote_profile_service.rb:4:3: C: Cyclomatic complexity for call is too high. [16/15]
  def call(body, account, resubscribe = false)
  ^^^
app/services/update_remote_profile_service.rb:4:3: C: Perceived complexity for call is too high. [16/10]
  def call(body, account, resubscribe = false)
  ^^^

251 files inspected, 13 offenses detected
```

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -81,11 +91,19 @@ Style/GuardClause:
...
+    '%i': '()'
+    '%w': '()'
+
 Style/PerlBackrefs:
   AutoCorrect: false
 
 Style/RegexpLiteral:
   Enabled: false
 
+Style/SymbolArray:
```
Comment (ykzts) [inline .rubocop.yml]: https://github.com/bbatsov/rubocop/pull/4144

File: .rubocop.yml [hunk 2]
```diff
@@ -17,12 +17,19 @@ Bundler/OrderedGems:
 Layout/AccessModifierIndentation:
   EnforcedStyle: indent
 
+Layout/EmptyLineAfterMagicComment:
```
Comment (ykzts) [inline .rubocop.yml]: https://github.com/bbatsov/rubocop/pull/3889

File: .rubocop.yml [hunk 3]
```diff
@@ -81,11 +91,19 @@ Style/GuardClause:
 Style/Lambda:
   Enabled: false
 
+Style/PercentLiteralDelimiters:
```
Comment (ykzts) [inline .rubocop.yml]: https://github.com/bbatsov/rubocop/pull/4231

**Review by ykzts — COMMENTED** [review]
- https://github.com/bbatsov/rubocop/pull/3825
- https://github.com/bbatsov/rubocop/pull/4133

**Review by clworld — APPROVED** [review]
OK

**Pull request merged**